contracts/root/RootChain.sol

```diff
@@ -22,50 +23,46 @@ contract RootChain is RootChainStorage, IRootChain {
         _;
     }
 
-    function submitHeaderBlock(
-        bytes calldata vote,
-        bytes calldata sigs,
-        bytes calldata txData
-    ) external {
-        RLPReader.RLPItem[] memory dataList = vote.toRlpItem().toList();
-        require(
-            keccak256(dataList[0].toBytes()) == heimdallId,
-            "Chain ID is invalid"
+    function submitHeaderBlock(bytes calldata data, bytes calldata sigs)
+        external
+    {
+        (
+            address proposer,
+            uint256 start,
+            uint256 end,
+            bytes32 rootHash,
+            bytes32 accountHash,
+            bytes32 _borChainID
+        ) = abi.decode(
+            data,
+            (address, uint256, uint256, bytes32, bytes32, bytes32)
         );
-        require(dataList[1].toUint() == VOTE_TYPE, "Vote type is invalid");
+        require(bytes32(CHAINID) == _borChainID, "Invalid bor chain id");
 
-        // validate hash of txData was signed as part of the vote
         require(
-            keccak256(dataList[4].toBytes()) ==
-                keccak256(abi.encodePacked(sha256(txData))),
-            "Extra data is invalid"
+            _buildHeaderBlock(proposer, start, end, rootHash),
+            "INCORRECT_HEADER_DATA"
         );
 
-        RLPReader.RLPItem[] memory extraData = txData.toRlpItem().toList();
-        extraData = extraData[0].toList();
-        RootChainHeader.HeaderBlock memory headerBlock = _buildHeaderBlock(
-            extraData
-        );
-        headerBlocks[_nextHeaderBlock] = headerBlock;
         // check if it is better to keep it in local storage instead
         IStakeManager stakeManager = IStakeManager(
             registry.getStakeManagerAddress()
         );
         // blockInterval, voteHash, stateRoot, sigs
         uint256 _reward = stakeManager.checkSignatures(
-            headerBlock.end.sub(headerBlock.start).add(1),
-            keccak256(vote),
-            bytes32(extraData[4].toUint()),
+            end.sub(start).add(1),
+            keccak256(abi.encodePacked(bytes(hex"01"), data)), // prefix 01 to data
```

Comment:
AFAIK validators can send data where prefix is 00 which means Noack(basically no vote but signed). so this is required to add inside the contract only. One thing we can do is just check for prefix instead of adding.

Reply:
Yeah I just asked to write the reason in solidity code as a comment.